superset-frontend/src/explore/actions/hydrateExplore.ts

```diff
@@ -97,6 +98,14 @@ export const hydrateExplore =
     }
 
     const initialDatasource = dataset;
+    initialDatasource.currency_formats = Object.fromEntries(
+      (initialDatasource.metrics ?? [])
+        .filter(metric => metric.currency !== undefined)
```

Comment:
probably relates to the earlier comment- this won't pick up null values, so maybe 
```suggestion
        .filter(metric => !!metric.currency)
```
would filter out null and undefined values. (and false and 0, etc if that's what you want)